velox/functions/lib/ArrayFunctions.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/functions/Macros.h"
+
+namespace facebook::velox::functions {
+
+/// This class implements the remove_nulls function of presto and array_compact
+/// function of spark.
+///
+/// DEFINITION:
+/// presto: remove_nulls(array(E)) -> array(E)
+/// spark: array_compact(array(E)) -> array(E)
+/// Remove all null elements from array. Returns NULL as result if array is
+/// NULL.
+template <typename T>
+struct ArrayRemoveNullFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // Fast path for primitives.
```

Comment:
Do we need this fast path as it provides the same implementation with the generic path? cc: @PHILO-HE

Reply:
I believe this is ok. ArrayTrimFunction has same implementation with generic type and primitive type

Reply:
@rui-mo, it's used in the existing code. I guess the perf. is slightly different with this path.

Reply:
@jinchengchenghh @zhli1142015 Do you have any idea how this path optimises the primitive types?

Reply:
registerArrayCompactFunction<int8_t>(prefix); requires the fast path, `Generic<T1>` can support any data type, usually it only register for complex data types, so we need it.

Reply:
Is this optimisation mainly for accelerating the registration?

Reply:
@jinchengchenghh any update? If this is good for accelerating the registration, I will keep this :)

Reply:
No, I don't thinks so. You may call array.push_back() for Generic type which is a different path with push_back int. @rui-mo

Reply:
I benchmarked the fast path and the generic path, and found 30% perf improvement with the fast path.